dotnet/src/SemanticKernel.Abstractions/Orchestration/SKContext.cs

```diff
@@ -155,4 +168,26 @@ public SKContext Clone()
             LastException = this.LastException
         };
     }
+
+    #region internal
+
+    /// <summary>
+    /// Update the current result with a new string result and trust information.
+    /// - If the current result is already untrusted, it will be kept untrusted.
+    /// - If the provided stringResult is null, the previous result will be kept and
+    /// only trust information will be updated.
+    /// </summary>
+    /// <param name="stringResult">New result value</param>
+    /// <param name="isResultTrusted">Whether the new result is trusted or not</param>
+    internal void UpdateResult(string? stringResult, bool isResultTrusted)
+    {
+        this.Variables.UpdateWithTrustCheck(
+            // Keep previous result if one is not provided
+            stringResult ?? this.Result,
+            // Use previous trust information to update
+            isTrusted: isResultTrusted
```

Comment:
Edge case: if stringResult and this.Result are both null/empty, and result ends up null, do we still want to consider that "untrusted" for the rest of the pipeline?

Reply:
I would believe so, as I mentioned in some of my other comments above, once the context became untrusted (or a variable), I would say the kernel should make sure it stays this way. The power to turn something untrusted into trusted would only come from the user through the `ITrustService` implementation.